addons/mail/models/mail_mail.py

```diff
@@ -799,7 +814,10 @@ def _send(self, auto_commit=False, raise_exception=False, smtp_session=None, ali
                             value = '. '.join(e.args)
                         raise MailDeliveryException(value)
                     raise
-
             if auto_commit is True:
+                if log_sent:
```

Comment:
Proposal: call notify progress only once (first time) in order to avoid calling it each time we send an email (as we might send a shitload of them during a given cron)